src/installer/installer.go

```diff
@@ -734,36 +734,62 @@ func (i *installer) cleanupInstallDevice() error {
 		return nil
 	}
 
-	vgName, err := i.ops.GetVGByPV(i.Device)
+	err := i.cleanupDevice(i.Device)
+
 	if err != nil {
 		return err
 	}
 
-	if vgName != "" {
-		i.log.Infof("A virtual group was detected on the installation device (%s) - cleaning", i.Device)
-		err = i.ops.RemoveVG(vgName)
+	if i.ops.IsRaidMember(i.Device) {
+		i.log.Infof("A raid was detected on the device (%s) - cleaning", i.Device)
+		var devices []string
+		devices, err = i.ops.GetRaidDevices(i.Device)
 
 		if err != nil {
 			return err
 		}
 
-		err = i.ops.RemovePV(i.Device)
+		for _, device := range devices {
+			// Cleaning the raid device itself before removing membership.
+			err = i.cleanupDevice(device)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		err = i.ops.CleanRaidMembership(i.Device)
 
 		if err != nil {
 			return err
 		}
 	}
 
-	if i.ops.IsRaidDevice(i.Device) {
-		i.log.Infof("A raid was detected on the installation device (%s) - cleaning", i.Device)
-		err = i.ops.CleanRaidDevice(i.Device)
+	return i.ops.Wipefs(i.Device)
+}
+
+func (i *installer) cleanupDevice(device string) error {
```

Comment:
This name is a bit too generic, can you change it to something along the lines of CleanLVM?

Reply:
This method could be extended to include not only lvm but every nested cleanup (relevant to installation devices and raid devices). Currently, this bug is regarding lvm, but it could also apply to any other nested cleanup.

Reply:
Such general function could simply call CleanLVM, and ideally wouldn't have all the cleanup code simply mixed in the same method body. 

Personally it confused me because I still don't really understand why you would consider RAID removal to not be "cleanup" but LVM to be "cleanup" .  There's clearly some distinction and one has to happen before the other so I don't think it's fair to give this a general name like "cleanupDevice" when it currently has a very specific cleanup purpose